pkg/reconciler/image/image.go

```diff
@@ -145,6 +151,15 @@ func (c *Reconciler) reconcileImage(ctx context.Context, image *buildapi.Image)
 		return image, nil
 	}
 
+	if image.Spec.NeedVolumeCache() && image.Spec.Cache.Volume.StorageClassName != "" {
+		_, err := c.StorageClassLister.Get(image.Spec.Cache.Volume.StorageClassName)
+		if err != nil {
```

Comment:
What is the reason to add this validation?

Reply:
@matthewmcnew @tomkennedy513 

I think this is an error that would only show up in the kpack-controller logs and result in a pending Image which is an error we should bubble up more

Reply:
could we add it to the status of the image

Reply:
> could we add it to the status of the image

That was my intention. I think it should be ready: false with a message

Reply:
ya I think that works. we should just make sure we document the required rbac change to the kpack controller

Reply:
> I think this is an error that would only show up in the kpack-controller logs and result in a pending Image which is an error we should bubble up more

I am not sure I understand. Won't this be written to the status and prevent future reonciles?

Reply:
I don't think the pvc will enter a failed state if the storage class does not exist, it just stays pending

Reply:
After pondering this for a few hours, I don't think it makes sense for the image reconciler to validate what is and is not a valid storageClass or persistent volume configuration. It seems that should be the responsibility of the PersistentVolumeClaim resource.

Reply:
removed!